conformance/tests/httproute-matching-combinations.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: matching-combinations
```

Comment:
I think it would be very helpful to have some tests covering a use case where multiple rules match the same request. For example

Rule A:
- Path match on /foo

Rule B:
- Query param match on animal=whale

Test: Request to `/foo?animal=whale`. 

You may already have some similar test cases in here that I'm missing, but if not, I think it would be really helpful to have something like this. This is conceptually similar to https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/tests/httproute-matching-across-routes.go, but would cover extended concepts like query param and method matching.

Reply:
Interesting. I completely missed this. Thanks Rob! I've added some to get started